lib/Frontend/SerializedDiagnosticConsumer.cpp

```diff
@@ -154,25 +154,41 @@ class SerializedDiagnosticConsumer : public DiagnosticConsumer {
       exitDiagBlock();
 
     // Write the generated bitstream to the file.
-    std::error_code EC;
-    std::unique_ptr<llvm::raw_fd_ostream> OS;
-    OS.reset(new llvm::raw_fd_ostream(State->SerializedDiagnosticsPath, EC,
-                                      llvm::sys::fs::F_None));
-    if (EC) {
-      // Create a temporary diagnostics engine to print the error to stderr.
-      DiagnosticEngine DE(SM);
-      PrintingDiagnosticConsumer PDC;
-      DE.addConsumer(PDC);
-      DE.diagnose(SourceLoc(), diag::cannot_open_serialized_file,
-                  State->SerializedDiagnosticsPath, EC.message());
+    std::unique_ptr<llvm::raw_fd_ostream> OS = openOutputFile();
+    if (!OS)
       return true;
-    }
 
     OS->write((char *)&State->Buffer.front(), State->Buffer.size());
     OS->flush();
     return false;
   }
 
+  bool informDriverThatCompilationCouldNotBeCompleted() const override {
+    // Hack! When the compiler ends abnormally, but there are no contents in
+    // serialized diagnostics
+    return openOutputFile() == nullptr;
```

Comment:
It is not obvious to me-pretending-to-read-the-code-without-context why you're comparing the result here to null and then throwing it away.

Reply:
Yeah, probably my and jordan's reviews here could both be addressed by replacing the comment with something like: "Open and immediately close a diagnostics output file, thereby creating a zero-sized file as a sentinel value to signal to the driver that this consumer was interrupted. Compare the resulting file to nullptr as a check for errors."

Reply:
@jordan Thanks for looking at this so soon! Great catch on the "not obvious" part. I've clarified the code. Will rename `informDriverThatCompilationCouldNotBeCompleted` per our offline conversation.